packages/nextcloud/test/api/notifications/notifications_helpers_test.dart

```diff
@@ -0,0 +1,146 @@
+      ),
+    );
+  });
+
+  group('DecryptedSubject', () {
+    DecryptedSubject createDecryptedSubject({
+      int? nid = 1,
+      String? app = 'app',
+      String? subject = 'subject',
+      String? type = 'type',
+      String? id = 'id',
+      bool? delete = false,
+      bool? deleteAll = false,
+    }) {
+      return DecryptedSubject((b) {
+        b
+          ..nid = nid
+          ..app = app
+          ..subject = subject
+          ..type = type
+          ..id = id
+          ..delete = delete
+          ..deleteAll = deleteAll;
+      });
+    }
+
+    group('constructor', () {
+      test('works correctly', () {
+        expect(
+          createDecryptedSubject,
+          returnsNormally,
+        );
+      });
+    });
+
+    test('supports value equality', () {
+      expect(
+        createDecryptedSubject(),
+        equalsBuilt(createDecryptedSubject()),
+      );
+
+      expect(
+        createDecryptedSubject().hashCode,
+        equals(createDecryptedSubject().hashCode),
+      );
+
+      expect(
+        createDecryptedSubject(),
+        isNot(equalsBuilt(createDecryptedSubject(deleteAll: true))),
+      );
+    });
+
+    group('rebuild', () {
+      test('returns the same object if not attributes are changed', () {
+        expect(
+          createDecryptedSubject().rebuild((_) {}),
+          equalsBuilt(createDecryptedSubject()),
+        );
+      });
+
+      test('replaces every attribute', () {
+        expect(
+          createDecryptedSubject().rebuild((b) {
+            b
+              ..nid = 2
+              ..app = 'new-app'
+              ..subject = 'new-subject'
+              ..type = 'new-type'
+              ..id = 'new-id'
+              ..delete = true
+              ..deleteAll = true;
+          }),
+          equalsBuilt(
+            createDecryptedSubject(
+              nid: 2,
+              app: 'new-app',
+              subject: 'new-subject',
+              type: 'new-type',
+              id: 'new-id',
+              delete: true,
+              deleteAll: true,
+            ),
+          ),
+        );
+      });
+    });
+
+    group('serialization', () {
+      test('fromJson works correctly', () {
+        expect(
+          DecryptedSubject.fromJson({
+            'nid': 1,
+            'app': 'app',
+            'subject': 'subject',
+            'type': 'type',
+            'id': 'id',
+            'delete': false,
+            'delete-all': false,
+          }),
+          equalsBuilt(createDecryptedSubject()),
+        );
+      });
+
+      test('toJson works correctly', () {
+        expect(
+          createDecryptedSubject().toJson(),
+          equals({
+            'nid': 1,
+            'app': 'app',
+            'subject': 'subject',
+            'type': 'type',
+            'id': 'id',
+            'delete': false,
+            'delete-all': false,
+          }),
+        );
+      });
+    });
+
+    test('to string', () {
```

Comment:
IMO this test is a bit unnecessary, toString() will never be called in production code and rarely while developing

Reply:
I like having it to verify every attribute is also contained in the string, but I agree that it isn't necessary.
Now we have the test so :woman_shrugging: